catho/quantum
# build(QTM-597): Build css and save css styles in jest snapshots

## Description
https://jirasoftware.catho.com.br/browse/QTM-599
https://jirasoftware.catho.com.br/browse/QTM-597

## Review guide
- [ ] Doc (yarn storybook)
- [ ] Build (in ssr-quantum-simulator)

NOTE: 

- Following the guidelines below, it is expected that Button snapshot tests will break due to the CSS styling that is applied to the component.

- In regressive testing, documentation and build testing, it is expected that the component will be rendered correctly. When inspecting the component on the page, it is possible to see the CSS styles being applied to the component, but they are not necessary working, as the styled components styles have a higher priority

## Before testing

1. Create a css file to style the Button component in the respective component folder.
Ex: /quantum/components/Button/Button.module.css

2. In the css file, do some styling.
Ex:

`.test {
    color: aquamarine;
}`

3. In the component, import and use the created styling.
Ex:

`
import styles from './Button.module.css';
...
<StyledButton className={styles.test}
/>
`

4. In the /quantum/components/index.css file, import the Button component styling created earlier.
Ex:
`@import "./Button/Button.module.css"`

## How to test the build

In quantum, run:
`
yarn

yarn build
`

In ssr-quantum-simulator, run:

`
yarn

yarn add ../quantum/dist

yarn dev
`

In the ssr-quantum-simulator/pages/_app.js file, import the quantum css:
`
import '@catho/quantum/index.css';
`

In your browser, go to http://localhost:3000/quantum

On the opened page, inspect the Button component example and verify that the component's css stylings were applied correctly.
